GameDesign/Assets/Scripts/ClickManager.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ClickManager : MonoBehaviour
+{
+    private BossAngerManager bossAngerManager;
+    // Start is called before the first frame update
+    void Start()
+    {
+        bossAngerManager = BossAngerManager.Instance;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (Input.GetMouseButtonDown(0))
+        {
+            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+            RaycastHit hit;
+
+            if (Physics.Raycast(ray, out hit) && hit.collider.gameObject == gameObject)
+            {
+                bossAngerManager.triggerEvent_CorrectlyClicked();
+            }
+            else
+            {
+                bossAngerManager.triggerEvent_MissClicked();
```

Comment:
Based on this logic it seems that every time I click, each object that wasn't clicked will trigger this (so if I have 10 characters and only one was clicked, the other 9 will send the event of not clicked). Can you make a test to see if this is true or false?